components/homme/src/theta-l/model_init_mod.F90

```diff
@@ -62,6 +64,49 @@ subroutine model_init2(elem,hybrid,deriv,hvcoord,tl,nets,nete )
       do t=1,timelevels
          elem(ie)%state%phinh_i(:,:,nlevp,t) = elem(ie)%state%phis(:,:)
       enddo
+
+      ! initialize reference states used by hyberviscosity
+#define HV_REFSTATES_V1
+#ifdef HV_REFSTATES_V0
+      elem(ie)%derived%dp_ref=0
+      elem(ie)%derived%phi_ref=0
+      elem(ie)%derived%theta_ref=0
+#endif
+#ifdef HV_REFSTATES_V1
+      ps_ref(:,:) = hvcoord%ps0 * exp ( -elem(ie)%state%phis(:,:)/(Rgas*TREF)) 
+      do k=1,nlev
+         elem(ie)%derived%dp_ref(:,:,k) = ( hvcoord%hyai(k+1) - hvcoord%hyai(k) )*hvcoord%ps0 + &
+              (hvcoord%hybi(k+1)-hvcoord%hybi(k))*ps_ref(:,:)
+      enddo
+      call set_theta_ref(hvcoord,elem(ie)%derived%dp_ref,elem(ie)%derived%theta_ref)
+      temp=elem(ie)%derived%theta_ref*elem(ie)%derived%dp_ref
+      call phi_from_eos(hvcoord,elem(ie)%state%phis,&
+           temp,elem(ie)%derived%dp_ref,elem(ie)%derived%phi_ref)
+#endif
+#ifdef HV_REFSTATES_V2
+      ! use Newton to solve for surface pressure
+      ! so that Cp theta_ref grad(exner_ref) + grad(phis) = 0
+      T1 = .0065*TREF*Cp/g ! = 191                                                            
+      T0 = TREF-T1         ! = 97                                                             
+      exner(:,:) =  exp( -kappa*elem(ie)%state%phis(:,:)/(Rgas*TREF))
+      do i=1,5
+         exner(:,:) = exner(:,:) - &
+              (elem(ie)%state%phis(:,:)/Cp + T0*log(exner(:,:))+T1*exner(:,:)-T1 ) &
+              / (T0/exner(:,:)+T1)
+      enddo
+      ps_ref(:,:) = (exner(:,:)**(1/kappa)) / p0
```

Comment:
ps_ref(:,:) = (exner(:,:)**(1/kappa))  *  p0 <--- multiply instead of divide?

Exner^{1/\kappa} = p_s/p0, so, p_s = Exner^{1/\kappa} * p0 ?